src/app_engine/apprtc.py

```diff
@@ -219,21 +219,20 @@ def get_room_parameters(request, room_id, client_id, is_initiator):
   # doesn't actually support HD modes.
   hd = request.get('hd').lower()
   if hd and video:
-    message = 'The "hd" parameter has overridden video=' + video
+    # Use HTML char code for quoutes to avoid JSON parsing issues.
+    message = 'The &#34;hd&#34; parameter has overridden video='
     logging.warning(message)
-    # HTML template is UTF-8, make sure the string is UTF-8 as well.
-    warning_messages.append(message.encode('utf-8'))
+    warning_messages.append(message)
   if hd == 'true':
     video = 'mandatory:minWidth=1280,mandatory:minHeight=720'
   elif not hd and not video and get_hd_default(user_agent) == 'true':
     video = 'optional:minWidth=1280,optional:minHeight=720'
 
   if request.get('minre') or request.get('maxre'):
-    message = ('The "minre" and "maxre" parameters are no longer ' +
+    message = ('The &#34;minre&#34; and &#34;maxre&#34; parameters are no longer ' +
```

Comment:
It is weird that you have to do this escape for warning_messages and  indicates there is probably a bug somewhere if this is needed.